lib/ecto/repo/preloader.ex

```diff
@@ -252,13 +252,20 @@ defmodule Ecto.Repo.Preloader do
 
     # If we are returning many results, we must sort by the key too
     query =
-      case card do
-        :many ->
+      case {card, query.combinations} do
+        {:many, [{kind, _} | []]} ->
```

Comment:
I'm only raising for `:many` because you can get by without a subquery for `:one` queries doing something like this: https://github.com/elixir-ecto/ecto/issues/3839#issuecomment-1050581380.

This means someone using combinations for a `:one` query won't get a nice error message if they don't use the trick above. But I'm thinking that's better than raising for valid queries or breaking people's code who are using the trick above.